pkg/topic/consumer_metadata_migration.go

```
@@ -0,0 +1,323 @@
+	Committed       bool                                 `json:"committed"`
+	InventorySHA256 string                               `json:"inventory_sha256"`
+	Records         []coordinator.ConsumerMetadataRecord `json:"records"`
+	Inventory       StorageInventory                     `json:"inventory"`
+}
+
+func ReadConsumerMetadataSelection(path string) (ConsumerMetadataSelection, error) {
+	file, err := os.Open(path) // #nosec G304 -- operator supplied maintenance input.
+	if err != nil {
+		return ConsumerMetadataSelection{}, fmt.Errorf("open consumer metadata selection: %w", err)
+	}
+	defer file.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `file.Close` is not checked

(errcheck)